include/eve/memory/soa_vector.hpp

```diff
@@ -0,0 +1,268 @@
+
+    //! Returns a pointer to the beginning
+    EVE_FORCEINLINE auto data()  { return eve::algo::unalign(aligned_begin()); }
+
+    //! Returns a constant pointer to the beginning
+    EVE_FORCEINLINE auto data()  const { return eve::algo::unalign(aligned_begin()); }
+
+    //! @brief Returns the value of the `i`th element of the container
+    //! @param i Index of the value to retrieve
+    EVE_FORCEINLINE auto get(std::size_t i) const { return eve::read(begin() + i); }
+
+    //! @brief Modify the value of the `i`th element of the container
+    //! @param i Index of the value to write
+    //! @param v Value to write
+    EVE_FORCEINLINE void set(std::size_t i, value_type const& v)
+    {
+      return eve::write(begin() + i, kumi::flatten_all(v));
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Iterators
+    //! @{
+    //==============================================================================================
+    //! Returns an aligned iterator to the beginning
+    EVE_FORCEINLINE auto aligned_begin()  { return aligned_begin_impl(*this); }
+
+    //! Returns an aligned iterator to the beginning
+    EVE_FORCEINLINE auto aligned_begin()  const { return aligned_begin_impl(*this); }
+
+    //! Returns a constant aligned iterator to the beginning
+    EVE_FORCEINLINE auto aligned_cbegin() const { return aligned_begin(); }
+
+    //! Returns an iterator to the beginning
+    EVE_FORCEINLINE auto begin()  { return eve::algo::unalign(aligned_begin()); }
+
+    //! Returns an iterator to the beginning
+    EVE_FORCEINLINE auto begin()  const { return eve::algo::unalign(aligned_begin()); }
+
+    //! Returns a constant iterator to the beginning
+    EVE_FORCEINLINE auto cbegin() const { return begin(); }
+
+    //! Returns an iterator to the end
+    EVE_FORCEINLINE auto end()  { return begin() + size(); }
+
+    //! Returns an iterator to the end
+    EVE_FORCEINLINE auto end()  const { return begin() + size(); }
+
+    //! Returns a constant iterator to the end
+    EVE_FORCEINLINE auto cend() const { return end(); }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Comparisons and ordering
+    //! @{
+    //==============================================================================================
+    friend auto operator<=>(soa_vector const& lhs, soa_vector const& rhs) = default;
```

Comment:
This is poor quality.
Check sizes and then use `eve::algo::equal` for equality and `eve::algo::mismatch` for less. 
Also, I'm not sure this supports custom equality overloads, but the `algo` should.

Reply:
Noted